tests/run/nogil_conditional.pyx

```diff
@@ -0,0 +1,242 @@
+# mode: run
+
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+
+def test(int x):
+    """
+    >>> test(0)
+    110
+    """
+    with nogil(True):
+        x = f_nogil(x)
+        with gil(True):
+            x = f_gil(x)
+    return x
+
+
+cdef int f_nogil(int x) nogil:
+    cdef int y
+    y = x + 10
+    return y
+
+
+cdef int f_gil(int x) with gil:
+    cdef int y
+    y = x + 100
+    return y
+
+
+cdef int with_gil_func() except 0 with gil:
```

Comment:
`except 0` is a bit dangerous since `0` is also the default success value to return from a `cdef int` function when there is no explicit `return x`.

Reply:
changed to -1.
Note that in tests/run/nogil.pyx it is still 0.

Reply:
Ah, thanks. I fixed these tests and also added a compiler warning for this case.